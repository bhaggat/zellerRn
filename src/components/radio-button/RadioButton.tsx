import React from 'react';
import {View, ViewProps, StyleSheet} from 'react-native';

type RadioButtonProps = ViewProps & {
  isChecked?: boolean;
  size?: number;
};

const RadioButton = ({
  isChecked,
  size = 20,
  style,
  ...props
}: RadioButtonProps) => (
  <View
    testID="radioButton"
    style={[
      styles.radioButton,
      {height: size, width: size, borderRadius: size / 2},
      isChecked && styles.checked,
      style,
    ]}
    {...props}>
    {isChecked && (
      <View
        style={{
          height: size / 2,
          width: size / 2,
          borderRadius: size / 4,
          backgroundColor: '#0370CE',
        }}
        testID="innerRadioButton"
      />
    )}
  </View>
);

const styles = StyleSheet.create({
  radioButton: {
    borderWidth: 1,
    borderColor: '#E2E8F0',
    alignItems: 'center',
    justifyContent: 'center',
  },
  checked: {
    borderColor: '#0370CE',
  },
});

export default RadioButton;
