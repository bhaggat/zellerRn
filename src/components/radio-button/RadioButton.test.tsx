import React from 'react';
import {render} from '@testing-library/react-native';
import RadioButton from './RadioButton';

describe('RadioButton Component', () => {
  describe('Snapshot', () => {
    test('matches', () => {
      const tree = render(<RadioButton />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('Matched Checked', () => {
      const tree = render(<RadioButton isChecked={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Render', () => {
    it('renders correctly when unchecked', () => {
      const {getByTestId, queryByTestId} = render(<RadioButton />);
      const outerRadioButton = getByTestId('radioButton');
      expect(outerRadioButton).toBeTruthy();
      const innerRadioButton = queryByTestId('innerRadioButton');
      expect(innerRadioButton).toBeNull();
    });
    it('renders correctly when checked', () => {
      const {getByTestId} = render(<RadioButton isChecked={true} />);
      const outerRadioButton = getByTestId('radioButton');
      expect(outerRadioButton).toBeTruthy();
      const innerRadioButton = getByTestId('innerRadioButton');
      expect(innerRadioButton).toBeTruthy();
    });
  });
});
