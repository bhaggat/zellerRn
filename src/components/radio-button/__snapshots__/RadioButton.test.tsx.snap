// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component Snapshot Matched Checked 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#E2E8F0",
        "borderWidth": 1,
        "justifyContent": "center",
      },
      {
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      },
      {
        "borderColor": "#0370CE",
      },
      undefined,
    ]
  }
  testID="radioButton"
>
  <View
    style={
      {
        "backgroundColor": "#0370CE",
        "borderRadius": 5,
        "height": 10,
        "width": 10,
      }
    }
    testID="innerRadioButton"
  />
</View>
`;

exports[`RadioButton Component Snapshot matches 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#E2E8F0",
        "borderWidth": 1,
        "justifyContent": "center",
      },
      {
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
  testID="radioButton"
/>
`;
